pkg/server/helpers.go

```diff
@@ -350,3 +353,38 @@ func getUserFromImage(user string) (*int64, string) {
 	// If user is a numeric uid.
 	return &uid, ""
 }
+
+// ensureImageExists returns corresponding metadata of the image reference, if image is not
+// pulled yet, the function will pull the image when ensure is true, and return error when
+// ensure if false.
+func (c *criContainerdService) ensureImageExists(ctx context.Context, ref string, ensure bool) (*metadata.ImageMetadata, error) {
+	imageID, err := c.localResolve(ctx, ref)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve image %q: %v", ref, err)
+	}
+	if imageID != "" {
+		meta, err := c.imageMetadataStore.Get(imageID)
+		if err == nil {
+			return meta, nil
+		}
+		if err != nil && !metadata.IsNotExistError(err) {
```

Comment:
Nit: no need to check `err != nil` because if err is `nil`, the code would've returned in line 368.